requests_toolbelt/multipart/decoder.py

```diff
@@ -143,7 +143,7 @@ def body_part(part):
             return BodyPart(fixed, self.encoding)
 
         def test_part(part):
-            return part != b'' and part != b'\r\n' and part[:4] != b'--\r\n'
+            return part != b'' and part != b'\r\n' and part[:4] != b'--\r\n' and part != '--'
```

Comment:
Let's rewrite this to be

```py
return (part != b'' and
        part != b'\r\n' and
        part[:4] != b'--\r\n' and
        part != b'--')
```

The last `b'--'` is actually import given we're a single-source Python 2 and 3 compatible library.